src/main/kotlin/com/openlattice/chronicle/services/studies/StudyService.kt

```diff
@@ -282,8 +280,15 @@ class StudyService(
     override fun isNotificationsEnabled(studyId: UUID): Boolean {
         logger.info("Checking notifications enabled on studyId = {}", studyId)
 
-        //TODO: Write SQL query to just retrieve notifications enabled field.
-        return getStudy(studyId).notificationsEnabled
+        return storageResolver.getPlatformStorage().connection.use { connection ->
+            connection.prepareStatement(GET_NOTIFICATION_STATUS_SQL).use { ps ->
+                ps.setObject(1, studyId)
+                ps.executeQuery().use { rs ->
+                    if (rs.next()) rs.getBoolean(NOTIFICATIONS_ENABLED.name)
```

Comment:
Are you sure about `rs.next()` vs `rs.hasNext()`?

Reply:
seems to work as intended